File: src/frogjumping.py
t = int(input())

def poschecker(a:int, b:int, k:int):
    if (a == b) and (k % 2 == 0):
        return 0
    elif (a == b) and (k % 2 == 1):
        return a
    else:
        if k % 2 == 0:
            pos = (a * int(k / 2)) - (b * int(k / 2)); return pos
        else:
            pos = (a * (int(k / 2) + 1)) - (b * int(k / 2)); return pos

for i in range(t):
    a, b, k = map(int, input().split())
    print(poschecker(a, b, k))

File: src/frogjumping.cpp
#include<bits/stdc++.h>
using namespace std;

long long int poschecker(long long int a, long long int b, long long int k){
    if((a == b) && (k % 2 == 0)){
        return 0;
    }
    else if((a == b) && (k % 2 == 1)){
        return a;
    }
    else{
        long long int pos = 0;
        if(k % 2 == 0){
            pos = (a * (k/2LL)) - (b * (k / 2LL)); return pos;
        }
        else{
            pos = (a * ((k / 2LL) + 1)) - (b * (k / 2LL)); return pos;
        }
    }
}

int main(){
    int t; cin >> t;
    while(t--){
        long long int a, b, k; cin >> a >> b >> k;
        long long int pos = 0;
        pos = poschecker(a, b, k);
        cout << pos << endl;
    }
    return 0;
}